main.py

```diff
@@ -1,15 +1,106 @@
+import json
+import os
+import sys
+import time
+import traceback
+
 from google.cloud import storage
 
-from honeyflare import process_bucket_object
+from honeyflare import (
+    create_libhoney_client,
+    process_bucket_object,
+    RetriableError,
+    logfmt,
+    vault,
+)
 
 storage_client = storage.Client()
-bucket = storage_client.bucket(os.environ['HONEYFLARE_BUCKET'])
+
+# Check for required envvars to fail early on invalid deployments
+honeycomb_dataset = os.environ.get('HONEYCOMB_DATASET')
+if honeycomb_dataset is None:
+    raise ValueError('Missing environment variable HONEYCOMB_DATASET')
+
+honeycomb_meta_dataset = os.environ.get('HONEYCOMB_META_DATASET')
+if honeycomb_meta_dataset is None:
+    raise ValueError('Missing environment variable HONEYCOMB_META_DATASET')
+
+honeycomb_key = os.environ.get('HONEYCOMB_KEY')
```

Comment:
How are these env variables passed in? Is there any reason we wouldn't use vault to also distribute things like `HONEYCOMB_KEY`?

Reply:
They are deployed with the function, in our case defined in the terraform module. Secrets stores are best used for secrets only, using them for other things makes it much harder to just inspect what values are set for general config.